src/main/java/frc/robot/subsystems/ChassisSubsystem.java

```diff
@@ -151,30 +152,41 @@ public void arcadeDrive(double speed, double rotation) {
    * @param scale
    */
   public void drive(RobotContainer robotContainer, DriverOI driverOI, int scale) {
+    String driveTrainType = Preferences.getString(Constants.DRIVE_TRAIN_TYPE, "DEFAULT");
     speedMultiplier = driverOI.getJoystick().getRawButton(Constants.RIGHT_BUMPER) ? crawl : normal;
     dir = driverOI.getJoystick().getRawButton(Constants.LEFT_BUMPER) ? Constants.FORWARD : Constants.REVERSE;
-    
-    RobotContainer.DriveTrainType driveTrainType = robotContainer.getDriveTrainType();
-
-    if (driveTrainType == RobotContainer.DriveTrainType.TANK) {
-      double rightVal = 0.0;  
-      double leftVal = 0.0;
-      if(dir == Constants.REVERSE) {
-        rightVal = getScaledValue(driverOI.getJoystick().getRawAxis(Constants.RIGHT_JOYSTICK_Y), scale, RobotSide.RIGHT);
-        leftVal = getScaledValue(driverOI.getJoystick().getRawAxis(Constants.LEFT_JOYSTICK_Y), scale, RobotSide.LEFT);
-      } else if(dir == Constants.FORWARD) {
-        rightVal = getScaledValue(driverOI.getJoystick().getRawAxis(Constants.LEFT_JOYSTICK_Y), scale, RobotSide.RIGHT);
-        leftVal = getScaledValue(driverOI.getJoystick().getRawAxis(Constants.RIGHT_JOYSTICK_Y), scale, RobotSide.LEFT);
-      }
-      SmartDashboard.putNumber("Left Speed", leftVal);
-      SmartDashboard.putNumber("Right Speed", rightVal);
-      tankDrive(leftVal * speedMultiplier * dir, rightVal * speedMultiplier * dir);
-    } else if (driveTrainType == RobotContainer.DriveTrainType.ARCADE) {
-      double speed = getScaledValue(driverOI.getJoystick().getRawAxis(Constants.LEFT_JOYSTICK_Y), scale, RobotSide.LEFT) * dir;
-      double rotation = getScaledValue(driverOI.getJoystick().getRawAxis(Constants.RIGHT_JOYSTICK_X), scale, RobotSide.RIGHT);
-      SmartDashboard.putNumber("Speed", speed);
-      SmartDashboard.putNumber("Rotation", rotation);
-      arcadeDrive(speed, rotation);
+
+    switch (driveTrainType) {
```

Comment:
This isn't simplify the code.. This is another way to write conditional statement. We don't really need a switch block here. There is really two cases here.  In addition, you are defaulting to Arcade drive.

Reply:
Take or leave any of the changes, your choice. But Arcade needs to be the default tonight.

Reply:
It is arguable that this version of changes is more maintainable. The other version can easy be changed as well. The current code is default to TANK by default. If the team was to change to Arcade default, there is only one line of code change. I don't see that this pull request is needed.